test/tests/functional/pbs_reservations.py

```diff
@@ -1123,3 +1123,409 @@ def test_reservations_with_expired_subjobs(self):
         # Check status of the parent-job using qstat -fx once reservation ends
         self.server.expect(JOB, {'job_state': 'F', 'substate': '91'},
                            extend='x', id=jid)
+
+    @skipOnCpuSet
+    def test_ASAP_resv_with_job_array(self):
+        """
+        Test job array with ASAP reservation
+        """
+        self.common_steps()
+
+        # Submit a short job to take up the resources (J1)
+        a = {'Resource_List.walltime': '20',
+             'Resource_List.select': '1:ncpus=4'}
+        j = Job(TEST_USER, attrs=a)
+        j.set_sleep_time(20)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, jid)
```

Comment:
why do we need to submit this job?

Reply:
Submitted this job to take up all resources , before submitting job-array .